unified_planning/engines/parallel.py

```diff
@@ -174,12 +187,14 @@ def _validate(
 
 def _run(
     idx: int,
-    EngineClass: type,
+    factory: "up.engines.factory.Factory",
+    engine_name: str,
     options: Dict[str, str],
     signaling_queue: Queue,
     fname: str,
-    *args
+    *args,
 ):
+    EngineClass = factory._engines[engine_name]
```

Comment:
This accesses a private field of the factory. Maybe we can make a public accessor method?